pkg/sql/schemaexpr/check_constraint_test.go

```diff
@@ -65,15 +65,15 @@ func TestCheckConstraintBuilder_Build(t *testing.T) {
 
 		// Allow expressions that result in a bool.
 		{"ck", "a", true, "a", "ck"},
-		{"ck", "b = 0", true, "b = 0", "ck"},
-		{"ck", "a AND b = 0", true, "a AND (b = 0)", "ck"},
+		{"ck", "b = 0", true, "b = 0:::INT8", "ck"},
+		{"ck", "a AND b = 0", true, "a AND (b = 0:::INT8)", "ck"},
```

Comment:
Why do only some expressions have type assertions?

Reply:
the serializable formatting flags will serialize all "potentially ambiguous" constants with type annotations -- see the `fmtDisambiguateDatumTypes` formatting flag.